types/n3/index.d.ts

```diff
@@ -13,87 +13,104 @@ import * as RDF from "rdf-js";
 import { EventEmitter } from "events";
 
 export interface Prefixes {
-    [key: string]: RDF.NamedNode;
+    [key: string]: NamedNode;
 }
 
-export class Term implements RDF.Term {
-    termType: "NamedNode" | "BlankNode" | "Literal" | "Variable" | "DefaultGraph";
-    id: string;
-    value: string;
-    constructor(iri: string);
-    toJSON(): string;
-    equals(other: RDF.Term): boolean;
-    static subclass(type: any): void;
-}
+export type Term = NamedNode | BlankNode | Literal | Variable | DefaultGraph;
 
-export class NamedNode extends Term implements RDF.NamedNode {
+export class NamedNode implements RDF.NamedNode {
     termType: "NamedNode";
     value: string;
     constructor(iri: string);
+    id: string;
+    toJSON(): {};
+    equals(other: Term): boolean;
+    static subclass(type: any): void;
 }
 
-export class BlankNode extends Term implements RDF.BlankNode {
+export class BlankNode implements RDF.BlankNode {
     static nextId: number;
     termType: "BlankNode";
     value: string;
     constructor(name: string);
+    id: string;
+    toJSON(): {};
+    equals(other: Term): boolean;
+    static subclass(type: any): void;
 }
 
-export class Variable extends Term implements RDF.Variable {
+export class Variable  implements RDF.Variable {
     termType: "Variable";
     value: string;
     constructor(name: string);
+    id: string;
+    toJSON(): {};
+    equals(other: Term): boolean;
+    static subclass(type: any): void;
 }
 
-export class Literal extends Term implements RDF.Literal {
+export class Literal implements RDF.Literal {
     static readonly langStringDatatype: NamedNode;
     termType: "Literal";
     value: string;
+    id: string;
+    toJSON(): {};
+    equals(other: Term): boolean;
+    static subclass(type: any): void;
     language: string;
-    datatype: RDF.NamedNode;
+    datatype: NamedNode;
     datatypeString: string;
     constructor(id: string);
 }
 
-export class DefaultGraph extends Term implements RDF.DefaultGraph {
+export class DefaultGraph implements RDF.DefaultGraph {
     termType: "DefaultGraph";
     value: "";
     constructor();
+    id: string;
+    toJSON(): {};
+    equals(other: Term): boolean;
+    static subclass(type: any): void;
 }
 
+export type Quad_Subject = NamedNode | BlankNode | Variable;
```

Comment:
Same comment here as in the RDFJS typings. The N3 implementation doesn't restrict the term type for S, P, O and G. So the typings should also allow any term type to be used for triple/quad components.